fastlane/Fastfile

```diff
@@ -424,16 +355,123 @@ platform :android do
     version = release_version_current
 
     # Wrap up
-    remove_branch_protection(repository: GH_REPOSITORY, branch: "release/#{version}")
-    set_milestone_frozen_marker(repository: GH_REPOSITORY, milestone: version, freeze: false)
-    create_new_milestone(repository: GH_REPOSITORY, need_appstore_submission: true)
-    close_milestone(repository: GH_REPOSITORY, milestone: version)
+    remove_branch_protection(repository: GITHUB_REPO, branch: "release/#{version}")
+    set_milestone_frozen_marker(repository: GITHUB_REPO, milestone: version, freeze: false)
+    create_new_milestone(repository: GITHUB_REPO, need_appstore_submission: true)
+    close_milestone(repository: GITHUB_REPO, milestone: version)
 
     push_to_git_remote(tags: false)
     trigger_release_build(branch_to_build: "release/#{version}")
     create_backmerge_pr
   end
 
+  # This lane publishes a release on GitHub and creates a PR to backmerge the current release branch into the next release/ branch
+  #
+  # @param [Boolean] skip_confirm (default: false) If set, will skip the confirmation prompt before running the rest of the lane
+  #
+  lane :publish_release do |skip_confirm: false|
+    ensure_git_status_clean
+    ensure_git_branch(branch: '^release/')
+
+    version_number = release_version_current
+
+    current_branch = "release/#{version_number}"
+    next_release_branch = "release/#{release_version_next}"
+
+    UI.important <<~PROMPT
+      Publish the #{version_number} release. This will:
+      - Publish the existing draft `#{version_number}` release on GitHub
+      - Which will also have GitHub create the associated git tag, pointing to the tip of the branch
+      - If the release branch for the next version `#{next_release_branch}` already exists, backmerge `#{current_branch}` into it
+      - If needed, backmerge `#{current_branch}` back into `#{DEFAULT_BRANCH}`
+      - Delete the `#{current_branch}` branch
+    PROMPT
+    UI.user_error!("Terminating as requested. Don't forget to run the remainder of this automation manually.") unless skip_confirm || UI.confirm('Do you want to continue?')
+
+    UI.important "Publishing release #{version_number} on GitHub"
+
+    publish_github_release(
+      repository: GITHUB_REPO,
+      name: version_number
+    )
+
+    # At this point, an intermediate branch has been created by creating a backmerge PR to a hotfix or the next version release branch.
+    # This allows us to safely delete the `release/*` remote branch.
+    remove_branch_protection(repository: GITHUB_REPO, branch: current_branch)
```

Comment:
ℹ️  I removed the call to `create_backmerge_pr` that was present back in the [original PR](https://github.com/Automattic/pocket-casts-android/pull/3382) via 7b80a1cffe4391aca19dfad4ded2758e4934d279 to address the feedback discussed in https://github.com/Automattic/pocket-casts-android/pull/3382#discussion_r1890699811